app/views/MessagesView/hooks/useMessages.tsx

```diff
@@ -0,0 +1,80 @@
+import { useState, useEffect } from 'react';
+
+import { Services } from '../../../lib/services';
+import { Encryption } from '../../../lib/encryption';
+import { IMessage, SubscriptionType } from '../../../definitions';
+
+interface IUseMessage {
+	rid: string;
+	t: SubscriptionType;
+	screenName: string;
+	userId: string;
+}
+
+export const useMessages = ({ rid, screenName, t, userId }: IUseMessage) => {
+	const [loading, setLoading] = useState(false);
+	const [messages, setMessages] = useState<IMessage[]>([]);
+	const [total, setTotal] = useState(-1);
+
+	const fetchFiles = async () => {
+		const result: any = await Services.getFiles(rid, t, messages.length);
+		if (result.success) {
+			result.messages = await Encryption.decryptFiles(result.files);
+			return result;
+		}
+	};
+
+	const fetchMessages = () => {
+		switch (screenName) {
+			case 'Files':
+				return fetchFiles();
+			case 'Mentions':
+				return Services.getMessages(rid, t, { 'mentions._id': { $in: [userId] } }, messages.length);
```

Comment:
I think we'll have to upgrade this code to match server 7.0 and staying backwards compatible.